src/Error/PhpError.php

```diff
@@ -0,0 +1,191 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.4.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Error;
+
+/**
+ * Object wrapper around PHP errors that are emitted by `trigger_error()`
+ */
+class PhpError
+{
+    /**
+     * @var int
+     */
+    private $code;
```

Comment:
Is this class meant to be final?

We could type the properties now that 4.4 is php 7.4+.

Reply:
> Is this class meant to be final?

No I just don't see a use case for allowing subclasses to read/write these properties directly when we have read accessor methods. I see PhpError objects as immutable. There are no useful scenarios I can see right now where we would ever need to mutate an instance.